iree_tests/download_remote_files.py

```diff
@@ -81,14 +83,26 @@ def download_azure_remote_file(test_dir: Path, remote_file: str):
         blob_size_str = human_readable_size(blob_properties.size)
         remote_md5 = get_remote_md5(remote_file, blob_properties)
 
-        local_file_path = test_dir / remote_file_name
+        cache_location = os.getenv("IREE_TEST_FILES", default="")
+        if cache_location == "":
+            os.environ["IREE_TEST_FILES"] = str(REPO_ROOT)
+            cache_location = REPO_ROOT
+        if cache_location == REPO_ROOT:
```

Comment:
String and Path handling here could be tricky. We might want to convert the string coming from the environment variable to a canonical path before handling it further. 

Could use https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve for that